expression/builtin_time.go

```diff
@@ -326,7 +326,7 @@ func (c *dateLiteralFunctionClass) getFunction(ctx sessionctx.Context, args []Ex
 	}
 	str := dt.GetString()
 	if !datePattern.MatchString(str) {
-		return nil, types.ErrWrongValue.GenWithStackByArgs(types.DateTimeStr, str)
+		return nil, types.ErrWrongValue.GenWithStackByArgs(strings.ToUpper(types.DateStr), str)
```

Comment:
```suggestion
		return nil, types.ErrWrongValue.GenWithStackByArgs(types.DateStr, str)
```

Reply:
I think there are two problems in this issue:
1. error information returns wrong date type
2. data type strings is typed in lowercase

For the first problem, modify the returned date type is ok.
For the second problem, I think we should modify the const strings (e.g. types.DateStr), instead of using strings.ToUpper in some places. Because there are many usage of the error const string.  But I think we shouldn't modify this, it will change many error message.

Reply:
Thanks for your kind comment. You mean there is no need to pay more attention to the case inconsistency between `date` and `DATE`, and lower case is just OK temporarily.

Correct me if i'm wrong, ths

Reply:
@yuqi1129 I agree with @wshwsh12 , as long as we provide enough information in the error message, there is no need to make it exact the same with MySQL.

Reply:
ok, i will update the PR as you two suggested, thanks